### ISSUE TITLE
Some Bugs in 3rd Person's view

### ISSUE THREAD

**Issue by wwar-XeStro** [minetest/minetest#10841]
##### Minetest versions: all versions i have tryed from 2016 to 5.3 im sure that worked before these versions
 third person camera has some x-ray vision, interact through walls bugs.

 

#1 

when using 3rd person's view in a stuck place you can see through the blocks, for example if you are underground and use that camera you will see all mines nearby, generally that works when you aimmyour camera to down. (that happens on all clients i've listed above).

![Image_2](https://user-images.githubusercontent.com/72745616/105355607-6f7c3380-5c03-11eb-987f-98d32724bbf5.png)

When using 3rd person's view on Computers: you get only 1 block reach, on phone devises you reach starts from the screen to 1 block infront of the character.

#2

When you use the 3rd person's back, and you are inside blocks but without a corner and aiming right between 2 of the blocks around you in the degrees between the blocks you will be able to see through blocks.

![image_1](https://user-images.githubusercontent.com/72745616/105355064-8ff7be00-5c02-11eb-8c5b-7394ea198b87.png)

#3

PC(aiming between blocks in the right edges may make you interact through blocks but you won't be able to see.)
On mobile clients as i said you have a reach from the camera to 1 block infront of the character so you can reach like the glass inside the red tringles:
![Image_3](https://user-images.githubusercontent.com/72745616/105356222-2ed0ea00-5c04-11eb-8c69-af1eaaf8853d.png)

If there are solid objects/blocks in the green "w" above we see like:
![Image_5](https://user-images.githubusercontent.com/72745616/105356621-af8fe600-5c04-11eb-8bb1-f29ec5905557.png)
when aiming like that you interact with the solid blocks i have placed above.

Possible fixes:

#1 => I don't know.
#2 => Making the corner degrees solid can fix that.
#3 => Making the 3rd person's back's interact reach start from the character to normal reach will fix that.

**Comment by wwar-XeStro** [comment]
This is the first issue i post on github, i hope i did it correctly, sorry if it isn't very understandable :)

**Comment by SmallJoker** [comment]
Duplicate of #2276. Same root cause: missing node samples to calculate the intersection. Proper fix in #9211 which turned out to be kinda slow.

**Comment by wwar-XeStro** [comment]
Well, i waited that to be fixed since 2016, these are abusable glitchs and can be helpful but whe they happen they make the game looks bad.. and i said that on some places and alot of people told me to open an issue on github

**Comment by wwar-XeStro** [comment]
> Duplicate of #2276. Same root cause: missing node samples to calculate the intersection. Proper fix in #9211 which turned out to be kinda slow.

You didn't mention that we can interact through blocks on #9211, and interacting is worse that only looking through them..

**Comment by SmallJoker** [comment]
Right. I did not consider that interacting is also a concern. I'll re-open this issue because it contains a more detailed description and additional error cases.

**Comment by wwar-XeStro** [comment]
Interaction is no longer possible after adding a cross hair into 3rd person view.
